### ISSUE TITLE
Some warnings are not reported when running clippy in Docker

### ISSUE THREAD

**Issue by Disasm** [rust-lang/rust-clippy#12837]
### Summary

Some warnings are not reported when clippy is executed inside Docker. The same warnings are reported normally on the host machine. I tried different compiler versions and different host machines. The issue is reproducible also with the latest stable compiler version.

### Reproducer

I tried this code:

```rust
pub fn output(value: u32) {
    println!("{}", format!("{value}").to_string());
}
```

Docker command: `docker run -ti -u $(id -u):$(id -g) -v $(pwd):/work --workdir=/work rust:1.78.0-slim-bullseye`.
Inside docker I run `rustup component add clippy` and then `cargo clippy`.

I expected to see this happen:
```
warning: `to_string` applied to a type that implements `Display` in `println!` args
   --> src/lib.rs:2:50
    |
2   |       println!("{}", format!("{value}").to_string());
    |  __________________________________________________^
3   | | }
...   |
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#to_string_in_format_args
    = note: `#[warn(clippy::to_string_in_format_args)]` on by default
```

Instead, this happened: clippy doesn't issue any warnings.

### Version

```text
rustc 1.78.0 (9b00956e5 2024-04-29)
binary: rustc
commit-hash: 9b00956e56009bab2aa15d7bff10916599e3d6d6
commit-date: 2024-04-29
host: x86_64-unknown-linux-gnu
release: 1.78.0
LLVM version: 18.1.2
(same compiler was used on the host and inside docker)
```

**Comment by Disasm** [comment]
I found that the difference between the environments is in the list of installed components. The warning is issued only when `rust-src` component is installed.

**Comment by Disasm** [comment]
Another example:
```rust
pub fn output(value: u32) {
    std::fs::write("test", format!("{value}").to_string()).unwrap();
}
```

```
warning: unnecessary use of `to_string`
 --> src/lib.rs:2:28
  |
2 |     std::fs::write("test", format!("{value}").to_string()).unwrap();
  |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_to_owned
  = note: `#[warn(clippy::unnecessary_to_owned)]` on by default
help: use
  |
2 ~     std::fs::write("test", &{
3 +         let res = $crate::fmt::format($crate::__export::format_args!($($arg)*));
4 +         res
5 ~     }).unwrap();
  |
```
The warning is not issued without the `rust-src` component.
